### PR TITLE
feat: add config model and runtime packages

### PR DESCRIPTION
**Pull request by JiaJunDeng5930**
## Summary
- add `selvedge-config-model` for config schema, defaults, and validation
- add `selvedge-config` for layered config loading, runtime patching, and persistence
- add TDD coverage for precedence, persistence, edge cases, and integration behavior

## Validation
- cargo fmt --all -- --check
- cargo clippy --all-targets --all-features -- -D warnings
- cargo test
- codex-review-final main

### CODE REVIEW

File: crates/config/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,605 @@
...
+    #[error("no active config file is selected for persistence")]
+    PersistUnavailable,
+    #[error("failed to persist config: {0}")]
+    PersistFailed(String),
+}
+
+fn resolve_explicit_path(path: PathBuf) -> Result<PathBuf, ConfigError> {
+    if path.is_file() {
+        return Ok(path);
+    }
```
[the remark this answers is not in the document]
Comment (JiaJunDeng5930) [reply crates/config/src/lib.rs]: Fixed in c6d4891. Explicit config paths are now canonicalized during initialization, and a regression test confirms persistence still writes back to the originally loaded file even after the process changes its working directory.

File: crates/config/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,605 @@
...
+where
+    F: FnOnce(&AppConfig) -> R,
+{
+    let global = CONFIG_SERVICE
+        .read()
+        .map_err(|_| ConfigError::LoadFailed("config service lock poisoned".to_owned()))?;
+    let service = global.as_ref().ok_or(ConfigError::NotInitialized)?;
+    let config = service.materialize_config()?;
+
+    Ok(reader(&config))
```
[the remark this answers is not in the document]
Comment (JiaJunDeng5930) [reply crates/config/src/lib.rs]: Fixed in c6d4891. `read()` now materializes the config through a helper and only invokes the caller callback after the global lock has been released. Added a unit test that verifies the global write lock is immediately available after materialization.
